cmake/sysbuild/suit.cmake

```diff
@@ -164,14 +163,23 @@ function(suit_create_package)
   set(CORE_ARGS)
   set(STORAGE_BOOT_ARGS)
 
-  sysbuild_get(CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION KCONFIG)
-  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
-  sysbuild_get(ENVELOPE_SHALL_BE_SIGNED IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  suit_set_absolute_or_relative_path(${SB_CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
+  set(ENVELOPE_SHALL_BE_SIGNED ${SB_CONFIG_SUIT_ENVELOPE_SIGN})
   if(NOT DEFINED ENVELOPE_SHALL_BE_SIGNED)
     set(ENVELOPE_SHALL_BE_SIGNED FALSE)
   endif()
 
+  list(APPEND CORE_ARGS
+    --core sysbuild,,,${CMAKE_BINARY_DIR}/zephyr/.config
+  )
+
   foreach(image ${IMAGES})
+    sysbuild_get(GENERATE_LOCAL_ENVELOPE IMAGE ${image} VAR CONFIG_SUIT_LOCAL_ENVELOPE_GENERATE KCONFIG)
```

Comment:
That indeed sounds like a problem in general, but I wouldn't be concerned in this particular case. The `CONFIG_SUIT_LOCAL_ENVELOPE_GENERATE` will be disabled only in very specific cases.

I'm not sure how this could be approached differently. This config determines if an artifact configuring a particular nRF54H20 domain is generated. In the flash companion use-case we build two application domain firmware, one is the main application the second is the flash companion.

We only need one envelope artifact per domain, so the main application has this Kconfig enabled, while the flash companion has it disabled.

The radio domain has its own envelope and it needs to be generated as well if a radiocore firmware is being built and flashed.